patchwise/utils/config.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from platformdirs import user_config_dir
+from patchwise import PACKAGE_PATH
+from pathlib import Path
+import yaml
+
+CONFIG_DIR = Path(user_config_dir())
+default_config_path = PACKAGE_PATH / "default_config.yaml"
+user_config_path = CONFIG_DIR / "patchwise_config.yaml"
+
+
+def read_from_config(path: Path) -> dict:
+    with open(path, "r") as file:
+        config_dict = yaml.safe_load(file)
+
+    if config_dict is None:
+        return {}
+
+    if not isinstance(config_dict, dict):
+        raise ValueError(f"Default config must be a dictionary, got {type(config_dict)}")
+
+    return config_dict
+
+
+def parse_config():
+    """
+    Parses both user and default configuration files and returns the union of the two with user taking precedence.
+    
+    """
+    default_options = read_from_config(default_config_path)
+    user_options = read_from_config(user_config_path)
```

Comment:
In the case of the user_config file not being present, PatchWise shouldn't fail, but in the case of default_config it should. We can & should keep the `read_from_config()` function the same, as you've done, we can just add a try except block here to specifically check for and ignore FileNotFoundError in the case of user_config.
```suggestion
    try:
        user_options = read_from_config(user_config_path)
    except FileNotFoundError:
        user_options: Dict[str, Any] = {}
```